temoa/utilities/visualizer.py

```
@@ -0,0 +1,311 @@
+        edge_key = f'{ic}-{oc}-{uuid.uuid4().hex[:8]}'
+        dg.add_edge(ic, oc, key=edge_key, **combined_attrs)
+
+    return dg
+
+
+def nx_to_vis(
+    nx_graph: GraphType,
+    output_filename: Path,  # Changed to Path object for clarity
+    html_title: str = 'NetworkX to vis.js Graph',
+    vis_options: dict[str, Any] | None = None,
+    show_browser: bool = True,
```

Ruff (0.14.1)

Boolean-typed positional argument in function definition

(FBT001)